server/channels/store/sqlstore/reaction_store.go

```diff
@@ -205,24 +207,90 @@ func (s *SqlReactionStore) DeleteAllWithEmojiName(emojiName string) error {
 	return nil
 }
 
-// DeleteOrphanedRows removes entries from Reactions when a corresponding post no longer exists.
-func (s *SqlReactionStore) DeleteOrphanedRows(limit int) (deleted int64, err error) {
-	// We need the extra level of nesting to deal with MySQL's locking
-	const query = `
-	DELETE FROM Reactions WHERE PostId IN (
-		SELECT * FROM (
-			SELECT PostId FROM Reactions
-			LEFT JOIN Posts ON Reactions.PostId = Posts.Id
-			WHERE Posts.Id IS NULL
-			LIMIT ?
-		) AS A
-	)`
-	result, err := s.GetMasterX().Exec(query, limit)
+func (s SqlReactionStore) PermanentDeleteByUser(userId string) error {
```

Comment:
This gets triggered when an admin deletes a user. The reason I felt it was in scope is because we don't have a way to truly clean up orphaned rows in the reactions table anymore. If we remove this we are opening ourselves to permanent orphaned reactions in the new workflow

Reply:
Hmm .. but correct me if I am wrong, this is completely new functionality right? As in, we never deleted reactions when an admin deleted a user right? Or did we?

Reply:
No we never did this before but we at least had the retention job to clean it up, now we won't

Reply:
Ahh okay I see my mistake. Previously, the retention job would clean up any danging reactions. Now it only cleans up those explicitly cleaned up by the parent task.

Ok, can we just use the original query ` DELETE FROM Reactions WHERE UserId = '<>';` for now? It's slow but deleting from the admin console is a relatively rare operation. Let's focus on the current scope of the PR for now and come back to this once it's merged.

Reply:
I don't think that solves our problem, even if we do `DELETE FROM Reactions WHERE UserId = '<>';`, we still have to get the impacted postIds and update the effected posts `has_reaction` field. So we may as well use `DELETE FROM Reactions WHERE PostId IN (...) AND UserId = '<>';` since it's faster. Should I just remove the batching from it and address the performance issues in another ticket? This still leaves us with the `UpdatePostHasReactionsOnDeleteQuery` issue that you pointed out though, I can't currently think of a way to batch those queries

Reply:
Ugh .. yeah you are right. Ok yeah, let's:
- remove batching
- Take all the postids in a single big slice.
- Do the DELETE in a single query.

So SELECT+DELETE runs in a single tx. This is important. Because we don't want to run the later individual UPDATE queries all part of the same tx, as the whole reactions table will be locked until all those small individual UPDATEs finish.

Then:
- Create a separate tx.
- Run the UPDATE queries by iterating the post slice again.

Then file a ticket to:
- Add batching reliably.
- Make the `UpdatePostHasReactionsOnDeleteQuery` batchable.

Reply:
Made these changes